README.md

<!-- badges: start -->

[![CRAN
status](https://www.r-pkg.org/badges/version/adeptdata)](https://cran.r-project.org/package=adeptdata)
[![](https://cranlogs.r-pkg.org/badges/grand-total/adeptdata)](https://cran.r-project.org/package=adeptdata)
[![](https://cranlogs.r-pkg.org/badges/last-month/adeptdata)](https://cran.r-project.org/package=adeptdata)
[![R-CMD-check](https://github.com/martakarass/adeptdata/workflows/R-CMD-check/badge.svg)](https://github.com/martakarass/adeptdata/actions)
<!-- badges: end -->

# adeptdata

Package `adeptdata` was created to host raw accelerometry data sets and
their derivatives. Some of them are used in the corresponding `adept`
package.

### Installation

Install from CRAN.

``` r
install.packages("adeptdata")
```

### Data objects

##### Object `acc_running`

The `acc_running` data frame is a sample raw accelerometry data
collected during 25 minutes of an outdoor run. Data were collected at
frequency 100 Hz with two ActiGraph GT9X Link sensors located at left
hip and left ankle. See `?acc_running` for details.

``` r
library(adeptdata)
head(acc_running)
#>     loc_id           date_time      x      y     z
#> 1 left_hip 2018-10-25 17:57:30 -0.199 -1.012 0.133
#> 2 left_hip 2018-10-25 17:57:30 -0.191 -1.027 0.133
#> 3 left_hip 2018-10-25 17:57:30 -0.191 -1.035 0.121
#> 4 left_hip 2018-10-25 17:57:30 -0.191 -1.027 0.109
#> 5 left_hip 2018-10-25 17:57:30 -0.191 -1.016 0.109
#> 6 left_hip 2018-10-25 17:57:30 -0.184 -1.012 0.121
lubridate::tz(acc_running$date_time)
#> [1] "UTC"
```

##### Object `acc_walking_IU`

The `acc_walking_IU` data frame is a sample of raw accelerometry data
collected during outdoor continuous walking from 32 healthy participants
between 23 and 52 years of age. Data were collected at frequency 100 Hz
simultaneously with four wearable accelerometers located at left wrist,
left hip and both ankles. See `?acc_walking_IU` for details.

``` r
head(acc_walking_IU)
#>      subj_id     loc_id time_s      x      y      z
#> 1 id86237981 left_wrist   0.00 -0.270 -0.773 -0.160
#> 2 id86237981 left_wrist   0.01 -0.266 -0.738 -0.172
#> 3 id86237981 left_wrist   0.02 -0.277 -0.707 -0.168
#> 4 id86237981 left_wrist   0.03 -0.277 -0.680 -0.160
#> 5 id86237981 left_wrist   0.04 -0.289 -0.656 -0.160
#> 6 id86237981 left_wrist   0.05 -0.316 -0.641 -0.164
```

##### Object `stride_template`

The `stride_template` is a list containing walking stride pattern
templates derived from accelerometry data collected at four body
locations: left wrist, left hip, left ankle, and right ankle. See
`?stride_template` for details.

``` r
template <- list(stride_template$left_ankle[[2]][1, ],
                 stride_template$left_ankle[[2]][2, ])
par(mfrow = c(1,2), cex = 1)
plot(template[[1]], type = "l", xlab = "", ylab = "", main = "Left ankle: template 1")
plot(template[[2]], type = "l", xlab = "", ylab = "", main = "Left ankle: template 2")
```

<img src="man/figures/README-unnamed-chunk-4-1.png" width="100%" />


NEWS.md
# adeptdata 1.0.1

* Added a `NEWS.md` file to track changes to the package.

# adeptdata 1.1

* Added a line `LazyDataCompression: xz` to DESCRIPTION per explicit instruction from CRAN. 


R/acc_running.R
#' @title Outdoor Run Raw Accelerometry Data
#'
#' @description Raw accelerometry data collected during 25 minutes of an
#' outdoor run. Data were collected at frequency 100 Hz simultaneously with
#' two wearable accelerometers located at left hip and left ankle.
#'
#' @format A \code{data.frame} with 300000 observations of 5 variables:
#' \itemize{
#'   \item \code{loc_id} - sensor location, one of: \code{"left_hip"}, \code{"left_ankle"},
#'   \item \code{date_time} - date and time of acceleration measurement collection stored as \code{POSIXct},
#'   \item \code{x} - acceleration measurement time-series collected from a "x" axis of the sensor accelerometer,
#'   \item \code{y} - acceleration measurement time-series collected from a "y" axis of the sensor accelerometer,
#'   \item \code{z} - acceleration measurement time-series collected from a "z" axis of the sensor accelerometer.
#' }
#'
#' @details
#' Data were collected during 25 minutes of an
#' outdoor run performed by an adult healthy female, 180 cm tall and of 67 kg weight
#'
#' The data were collected at frequency 100 Hz with two
#' ActiGraph GT9X Link sensors. One of the sensors was attached to the shoe with a clip
#' on the outside side of a left foot, just below the left ankle. The other sensor
#' was attached to the elastic belt located at hip, on the left side of a hip.
#'
#' Based on a mobile tracking application output, the ground elevation difference between
#' start and end point of the data collection is approximately 36 m (17 m at
#' the start point, 53 m at the finish point). The distance covered is approximately 3.35 km.
#'
#' The person from which the data were collected is Marta Karas, a co-author of the package.
#' The IRB Office Determination Request Form for Primary (New) Data Collection
#' request form was submitted in regard to the collection and further publishing of
#' these data. Based on preliminary review of the
#' request form submitted, it was determined that the data collection and further data publishing
#' activity described in the determination request does not qualify as human subjects research as
#' defined by DHHS regulations 45 CFR 46.102, and does not require IRB oversight.
#'
#'
"acc_running"


R/stride_template.R
#' @title Walking Stride Pattern Templates
#'
#' @description Walking stride pattern templates derived from raw accelerometry
#' data collected at four body locations: left wrist, left hip, left ankle,
#' and right ankle.
#'
#' @format A \code{list} with four named elements:
#' \itemize{
#'   \item \code{left_wrist},
#'   \item \code{left_hip},
#'   \item \code{left_ankle},
#'   \item \code{right_ankle}.
#' }
#' Each of the above is a five-element \code{list} of \code{matrix} objects.
#' The matrices are collection of (sub)population-specific stride pattern templates.
#' For example,
#' \itemize{
#'   \item \code{stride_template$left_wrist[[1]]} is a \code{1 x 200} \code{matrix}
#'   of one population-specific stride template derived from accelerometry data
#'   collected at left wrist.
#'   \item \code{stride_template$left_wrist[[2]]} is a \code{2 x 200} \code{matrix}
#'   of two distinct subpopulation-specific stride templates derived from accelerometry data
#'   collected at left wrist. Each row is a one subpopulation-specific stride template.
#'   \item \code{stride_template$right_ankle[[5]]} is a \code{5 x 200} \code{matrix}
#'   of five distinct subpopulation-specific stride templates derived from accelerometry data
#'   collected at right ankle.
#' }
#'
#' @details
#' Raw accelerometry data used to derive walking stride pattern templates
#' were collected as a part of the study on Identification of Walking,
#' Stair Climbing, and Driving Using Wearable Accelerometers, sponsored by the Indiana
#' University CTSI grant and conducted at the Department of Biostatistics,
#' Fairbanks School of Public Health at Indiana University. The study was led by Dr.
#' Jaroslaw Harezlak, assisted by Drs. William Fadel and Jacek Urbanek.
#' Study enrolled 32 healthy participants between 23 and 52 years of age.
#' Participants were asked, among others, to perform self-paced, undisturbed,
#' outdoor walking on the sidewalk. Accelerometry data were collected
#' at four body locations: left wrist, left hip, left ankle,
#' and right ankle.
#'
#' To derive empirical stride pattern, firstly, from each body location,
#' 642 data segments corresponding to individual walking strides
#' were manually segmented. Secondly, Vector Magnitude (VM), which is a univariate summary
#' of three-dimensional time-series of raw accelerometry data, was computed.
#' Lastly, 642 univariate vectors of VM were interpolated to have the same vector length,
#' scaled, and clustered into 1,2,3,4 and 5 clusters via correlation clustering.
#' The vectors obtained as point-wise means within each cluster
#' were defined to be subpopulation-specific stride pattern templates, respectively.
#'
"stride_template"


R/acc_walking_IU.R
#' @title Outdoor Continuous Walking Raw Accelerometry Data
#'
#' @description Raw accelerometry data collected during
#' outdoor continuous walking from 32 healthy participants between 23 and 52 years of age.
#' Data were collected at frequency 100 Hz simultaneously with
#' four wearable accelerometers located at left wrist, left hip and both ankles.
#'
#' @format A \code{data.frame} with 2590448 observations of 6 variables:
#' \itemize{
#'   \item \code{subj_id} - study participant ID,
#'   \item \code{loc_id} - sensor location, one of: \code{"left_wrist"}, \code{"left_hip"}, \code{"left_ankle"}, \code{"right_ankle"},
#'   \item \code{time_s} - duration of recorded exercise for a study participant, expressed in seconds,
#'   \item \code{x} - acceleration measurement time-series collected from a "x" axis of the sensor accelerometer,
#'   \item \code{y} - acceleration measurement time-series collected from a "y" axis of the sensor accelerometer,
#'   \item \code{z} - acceleration measurement time-series collected from a "z" axis of the sensor accelerometer.
#' }
#'
#' @details
#' Raw accelerometry data of continuous walking were collected as a part of the
#' study on Identification of Walking,
#' Stair Climbing, and Driving Using Wearable Accelerometers, sponsored by the Indiana
#' University CTSI grant and conducted at the Department of Biostatistics,
#' Fairbanks School of Public Health at Indiana University. The study was led by Dr.
#' Jaroslaw Harezlak, assisted by Drs. William Fadel and Jacek Urbanek.
#' The study was approved by the IRB of Indiana University;
#' all participants provided written informed consent.
#' Attached data set is anonymized.
#'
#' Study enrolled 32 healthy participants between 23 and 52 years of age.
#' Participants were asked, among others, to perform self-paced, undisturbed,
#' outdoor walking on the sidewalk. Accelerometry data were collected
#' simultaneously at four body locations: left wrist, left hip, left ankle,
#' and right ankle, at frequency 100 Hz. Duration time of outdoor walking exercise
#' ranges between 2,5 to 4 minutes for study participants.
#'
"acc_walking_IU"
